src/index.js

```diff
@@ -1447,5 +1452,266 @@ module.exports = {
         return expect.shift(parseXml(subject));
       }
     );
+
+    function scoreElementAgainstSpec(element, spec) {
+      const isTextMatching = (value, valueSpec) => {
+        if (typeof valueSpec === 'string') {
+          if (value === valueSpec) {
+            return true;
+          }
+        } else if (valueSpec instanceof RegExp) {
+          if (valueSpec.test(value)) {
+            return true;
+          }
+        } else if (typeof valueSpec === 'function') {
+          return true;
+        }
+
+        return false;
+      };
+      const isHtml = isInsideHtmlDocument(element);
+
+      let score = 0;
+
+      const nodeName = isHtml
+        ? element.nodeName.toLowerCase()
+        : element.nodeName;
+
+      if (isTextMatching(nodeName, spec.name)) {
+        score++;
+      }
+
+      if (isTextMatching(element.textContent, spec.textContent)) {
```

Comment:
Should we be fuzzy about leading and trailing whitespace here? I don't remember exactly how we're handling that in the other assertions.

Reply:
Let's trim the white space for all cases.